docs/DeveloperGuide.md

```diff
@@ -1,31 +1,400 @@
-# Developer Guide
+# Developer Guide for E-Duke-8
 
-## Design & implementation
+- [Developer Guide for E-Duke-8](#developer-guide-for-e-duke-8)
+  - [1. Introduction](#1-introduction)
+    - [1.1. Software Overview](#11-software-overview)
+    - [1.2. Setting up](#12-setting-up)
+  - [2. Design & implementation](#2-design--implementation)
+    - [2.1. Architecture](#21-architecture)
+    - [2.2. Model Component](#22-model-component)
+      - [2.2.1. Design of TopicList](#221-design-of-topiclist)
+      - [2.2.2. Implementation of TopicList](#222-implementation-of-topiclist)
+      - [2.2.3. Implementation of Notes](#223-implementation-of-notes)
+      - [2.2.4. Design of Option and OptionList](#224-design-of-option-and-optionlist)
+    - [2.3. Logic Component](#23-logic-component)
+      - [2.3.1. Design of Parser](#231-design-of-parser)
+      - [2.3.2. Implementation of MenuParser](#232-implementation-of-menuparser)
+      - [2.3.3. Design of QuizQuestionsManager](#233-design-of-quizquestionsmanager)
+      - [2.3.4. Implementation of QuizQuestionsManager](#234-implementation-of-quizquestionsmanager)
+      - [2.3.5. Design of Stats Feature](#235-design-of-stats-feature)
+    - [2.4. Storage Component](#24-storage-component)
+      - [2.4.1. Design of TopicsStorage](#241-design-of-topicsstorage)
+      - [2.4.2. Implementation of TopicsStorage](#242-implementation-of-topicsstorage)
+      - [2.4.3. Design of UserStorage](#243-design-of-userstorage)
+      - [2.4.4. Implementation of UserStorage](#244-implementation-of-userstorage)
+    - [2.5. UI Component](#25-ui-component)
+      - [2.5.1. Implementation of Ui](#251-implementation-of-ui)
+  - [3. Product scope](#3-product-scope)
+    - [3.1. Target user profile](#31-target-user-profile)
+    - [3.2. Value proposition](#32-value-proposition)
```

Comment:
Could all the key words in these headings be capitalised to be the same as the other headings?

Reply:
Alright will change that, thanks!